pkg/connection/store/vault/store.go

```diff
@@ -63,7 +63,7 @@ type SecretStore struct {
 }
 
 // NewSecretStore returns a new Vault SecretStore.
-func NewSecretStore(ctx context.Context, kube client.Client, cfg v1.SecretStoreConfig) (*SecretStore, error) {
+func NewSecretStore(ctx context.Context, kube client.Client, cfg v1.SecretStoreConfig) (*SecretStore, error) { // nolint: gocyclo
```

Comment:
Nit: Maybe add a comment about this linter being disabled? I'm guessing the function just went a little over the limit when you added the new switch.